space-concordia-robotics/robotics-prototype
# Serial init 207

# Assignee Section

## Description
A huge factor of how the communication is initialized across devices. It used to be super messy and now its refactored in much smaller code portion

- Enforced coding standards
- Split init_serial

closes #207 

## Testing

(You need a teensy 3.6)
1. Upload `Arm.ino` to the teensy
2. Run `roscore`
3. In a new terminal run `rosrun mcu_control ArmNode.py`. Observe that the armnode searches for the mcu and finds it
4. Run `rosrun mcu_control RoverNode.py` (Launch rover node with arm code on). This should fail to find the arm teensy (Current behavior is that it won't find it but it will still read from the input and will log whatever it gets, imo this should maybe be changed in another PR)
5. Upload `MobilePlatform` to the teensy
6. `rosrun mcu_control RoverNode.py`
7. Observe the shitty spam of voltage, but if you see what happened before, you should see ASTRo MCU Identified!
8. Run `rosrun mcu_control RoverNode.py uart`, it should tell you it couldnt find any UART devices
9. Upload `Science.ino` to the teensy
10. Run `rosrun mcu_control ScienceNode.py`. It should indicate that it succesfully connected.

Approval from at least one software team lead is necessary before merging.

# Reviewer Section

Aside from local testing and the General Integration Test it is implied that static analysis should be included in the verification process.


For Pull Requests that do not include code changes, it is not required to perform the tests above.
